firmware/main.c

```diff
@@ -99,13 +105,18 @@ uchar usbFunctionSetup(uchar data[8]) {
 
 		/*
 		 * Auto sck probing is done here vs in FUNC_ENABLEPROG
-		 * So that host applications that use FUNC_TRANSMIT instead to
+		 * So that host applications that use FUNC_TRANSMIT to do
 		 * their own raw ISP port access instead of letting the USBasp device
-		 * do it form them, will also benefit from the auto sck probing.
+		 * do it for them will also benefit from the auto sck probing.
 		 */
 
 		if (prog_sck == USBASP_ISP_SCK_AUTO) {
-			prog_sck = autosck(USBASP_ISP_SCK_1500);
+			/*
```

Comment:
With the existing avrdude, there is no way to tell it (avrdude) that the CONNECT command failed since the old CONNECT didn't return any parameters and it was assumed that would always work.
Luckily avrdude doesn't check for the number of returned bytes for that command. I guess it might be better to change the code to set the clock to a slow clock AND fail the command using the new style return status. The old avrdude would not look and blaze on, just like it does now but the new avrdude would see that the CONNECT with auto SCK failed.
This is probably better than what it is doing now.
Let me know if you want this change and I'll go do it. It is trivial. It won't affect the current avrdude but avrdude with the patch to look at the return sck  will fail appropriately when the connect fails.